### DISCUSSION TITLE
[Announcement] clarification on constraint-based search

### DISCUSSION THREAD

**Discussion by KihongHeo** [prosyslab-classroom/cs424-program-reasoning#361] [Announcements]
Hi all,

Here is the clarification regarding the last discussion on SMT encoding.

Your point is correct. The implication might produce non-sensical solutions. So the correct encoding should consist of conductions. 

For multiple input-output pairs, we separately encode their data-flow connections. Suppose we have two input-output pairs $(I, O), (I', O')$. Here is a portion of data-flow encoding:
$(l_{I} = l_{I_1} \implies I = I_1) \land (l_{I'} = l_{I_1} \implies I' = I_1) \cdots$

For details, see pages 14 and 15 of the updated slides. Let me clarify more next week.

Thanks for pointing out this one. Since it is quite important, we will give 1% bonus points to the students who opened the discussion (@kingdoctor123 , @retroinspect, @Allgot) . @Hhro please add bonus points.

**Comment by Hhro** [comment]
Ok, I added a 1% bonus to them.
